charts/timescaledb-single/templates/job-update-patroni.yaml

```diff
@@ -0,0 +1,57 @@
+# This file and its contents are licensed under the Apache License 2.0.
+# Please see the included NOTICE for copyright information and LICENSE for a copy of the license.
+{{/*
+The sole purpose of this job is to trigger Patroni to rewrite the
+Dynamic Configuration[1] in the DCS.
+
+As the configuration may change between every invocation of Helm, we
+need to run this trigger after every update.
+
+To prevent Helm failing to update a deployment, we add some random characters
+to the created job; Helm cannot create a job with the same name,
+however, (completed) jobs will not immediately be removed by Kubernetes.
+
+1: https://patroni.readthedocs.io/en/latest/dynamic_configuration.html
+*/}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}-patroni-{{ randAlphaNum 2 | lower }}"
+  labels:
+    app: {{ template "timescaledb.fullname" . }}
+    chart: {{ template "timescaledb.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    cluster-name: {{ template "clusterName" . }}
+  annotations:
+    "helm.sh/hook": post-upgrade
+    "helm.sh/hook-weight": "-5"
```

Comment:
for learning: what does this mean?

Reply:
> Tiller sorts hooks by weight and by name(for those hooks with the same weight) in ascending order. For example, -5, 0, 5 are executed in this order.

I guess it's not strictly needed here, so I may remove it